src/Data/Tree/AVL/Deletion.hs

```diff
@@ -1,8 +1,9 @@
 
-{-# LANGUAGE MultiWayIf     #-}
-{-# LANGUAGE NamedFieldPuns #-}
-{-# LANGUAGE LambdaCase #-}
-{-# LANGUAGE ExplicitForAll #-}
+{-# LANGUAGE ExplicitForAll      #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE LambdaCase          #-}
```

Comment:
LambdaCase should go into the global config (cabal).
https://github.com/serokell/serokell-util/blob/master/serokell-style.md#default-extensions

I'd also put  `FlexibleContexts` there as well. 

This is just a nit though.